src/test/java/athena/commands/DeleteCommandTest.java

```diff
@@ -0,0 +1,92 @@
+package athena.commands;
+
+import athena.Importance;
+import athena.TaskList;
+import athena.Ui;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeEach;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+class DeleteCommandTest {
+    private TaskList taskList;
+    private TaskList taskListWithoutTask;
+
+
+    public static TaskList getTaskList() {
+        TaskList taskList = new TaskList();
+        taskList.addTask("Assignment 1", "4pm", "2 hrs", "6pm", "12-12-2020",
+                Importance.HIGH, "Tough assignment", 1);
+        taskList.addTask("Assignment 2", "4pm", "2 hrs", "6pm", "13-12-2020",
+                Importance.MEDIUM, "Tough assignment", 1);
+        return taskList;
+    }
+
+    public static TaskList getTaskListWithoutTask() {
+        TaskList taskList = new TaskList();
+        taskList.addTask("Assignment 1", "4pm", "2 hrs", "6pm", "12-12-2020",
+                Importance.HIGH, "Tough assignment", 1);
+        return taskList;
+    }
+
+    @BeforeEach
+    public void setup() {
+        taskList = getTaskList();
+        taskListWithoutTask = getTaskListWithoutTask();
+    }
+
+    @Test
+    public void execute_validIndex_taskIsDeleted() {
+        assertDeletionSuccessful(2, taskList, taskListWithoutTask);;
+    }
+
+    @Test
+    public void execute_invalidIndex_returnsInvalidIndexMessage() {
+        assertDeletionFailsDueToInvalidIndex(-1, taskList);
+    }
+
+    /**
+     * Creates a new delete command.
+     *
+     * @param targetVisibleIndex of the person that we want to delete
+     */
+    private DeleteCommand createDeleteCommand(int targetVisibleIndex) {
+        DeleteCommand command = new DeleteCommand(targetVisibleIndex);
+        return command;
+    }
+
+    /**
+     * Executes the command, and checks that the execution was what we had expected.
+     */
+    private void assertCommandBehaviour(DeleteCommand deleteCommand, String expectedMessage,
+                                        TaskList expectedTaskList, TaskList actualTaskList) {
+        Ui ui;
+        Command result = DeleteCommand.execute(taskList, ui);
+        assertEquals(expectedMessage, result);
```

Comment:
Currently we don't return the printed message as a string so I think we can leave this line out for now. We should later think about how we want to return the printed string for testing purposes.